cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -2979,6 +2980,7 @@ struct AwsInstance {
   Aws::SDKOptions aws_options_;
   std::atomic<bool> is_initialized_;
```

Comment:
Can we remove `std::atomic<>`?

```suggestion
  bool is_initialized_;
```

Reply:
I think couldn't cuz is_initialized_ is exposed in AwsInstance::IsInitialized and in use.